// @framer/tryjohnny.sketchfab/code/SketchFab.tsx
import * as React from "react";
import { Animatable, PropertyControls, ControlType } from "framer"

interface Props {
    width: number;
    height: number;
    sketchfab: string;
    autostart: string;
    spinauto: boolean;
    entrance: boolean;
    scrollwheel: boolean;
}

export class SketchFab extends React.Component {
    static defaultProps = {
        width: 375,
        height: 196,
        sketchfab: "506d8fb15f354356ab3284c26a91862b",
        autostart: true,
        spinauto: "False",
        entrance: true,
        scrollwheel: true,
    };

    static propertyControls: PropertyControls<Props> = {
        sketchfab: { type: ControlType.String, title: "ID" },
        autostart: { type: ControlType.Boolean, title: "Autostart?" },
        entrance: { type: ControlType.Boolean, title: "Entrance?" },
        scrollwheel: { type: ControlType.Boolean, title: "Scrollwheel?" },
        spinauto: { type: ControlType.SegmentedEnum, title: "AutoSpin?", options: ["True","False"] }
    }

    render() {
        const width = Animatable.getNumber(this.props.width)
        const height = Animatable.getNumber(this.props.height)

        if (this.props.spinauto === "True") {
            var url = `https://sketchfab.com/models/${this.props.sketchfab}/embed?autostart=${this.props.autostart}&camera=${this.props.entrance}&scrollwheel=${this.props.scrollwheel}&autospin=0.5`;
        }
        else if (this.props.spinauto === "False") {
            var url = `https://sketchfab.com/models/${this.props.sketchfab}/embed?autostart=${this.props.autostart}&camera=${this.props.entrance}&scrollwheel=${this.props.scrollwheel}&autospin=0`;
        }

        return (
            <iframe width={width} height={height} frameBorder={0} src={url} />
        )
    }
}
