// @framer/toddjreynolds.elements-ui-kit/code/ElementsGrouping.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType, Stack } from "framer"
import { colors } from "./canvas"

// Learn more: https://framer.com/api

export function ElementsGrouping(props) {
    const { item1, item2, item3, item4, item5, ...rest } = props

    return (
        <Stack
            {...rest}
            gap={0}
            distribution="start"
            paddingTop={8}
            paddingBottom={8}
            borderRadius={30}
        >
            {item1}
            {item2}
            {item3}
            {item4}
            {item5}
        </Stack>
    )
}

ElementsGrouping.defaultProps = {
    height: "auto",
    width: 343,
    background: colors.TileBackground,
}

// Learn more: https://framer.com/api/property-controls/
addPropertyControls(ElementsGrouping, {
    item1: {
        title: "First",
        type: ControlType.ComponentInstance,
    },
    item2: {
        title: "Second",
        type: ControlType.ComponentInstance,
    },
    item3: {
        title: "Third",
        type: ControlType.ComponentInstance,
    },
    item4: {
        title: "Fourth",
        type: ControlType.ComponentInstance,
    },
    item5: {
        title: "Fifth",
        type: ControlType.ComponentInstance,
    },
})
