// @framer/toddjreynolds.elements-ui-kit/code/InlineButton.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType, Stack } from "framer"
import { Icon, iconNames } from "./Icon"
// Learn more: https://framer.com/api

export function InlineButton(props) {
    const { text, tint, onTap, ...rest } = props

    return (
        <Frame
            {...rest}
            background="null"
            onTap={onTap}
            style={{
                color: props.tint,
                fontSize: 17,
                fontWeight: 500,
            }}
        >
            <Stack
                direction="horizontal"
                alignment="center"
                distribution="center"
                background="transparent"
                width="100%"
                height="100%"
                gap={8}
            >
                <Frame height={24} width={24} background="null">
                    <Icon name={props.name} tintColor={props.tint} />
                </Frame>
                <Frame
                    height={24}
                    width="auto"
                    background="null"
                    style={{
                        paddingTop: 2,
                    }}
                >
                    <span>{text}</span>
                </Frame>
            </Stack>
        </Frame>
    )
}

InlineButton.defaultProps = {
    height: 56,
    width: 375,
    text: "Savannah",
    tint: "#4af196",
    name: "Add",
}

// Learn more: https://framer.com/api/property-controls/
addPropertyControls(InlineButton, {
    name: {
        type: ControlType.Enum,
        title: "Icon",
        options: iconNames.sort(),
    },
    text: {
        title: "Text",
        type: ControlType.String,
        defaultValue: "Hello Framer!",
    },
    tint: {
        title: "Tint",
        type: ControlType.Color,
        defaultValue: "#0099ff",
    },
    onTap: {
        type: ControlType.EventHandler,
    },
})
