// @framer/tryjohnny.better-youtube/code/Better_YouTube.tsx
import * as React from "react";
import { Animatable, PropertyControls, ControlType } from "framer"

interface Props {
    width: number;
    height: number;
    youtubeID: string;
    autoplay: boolean;
    controls: string;
    mute: boolean;
    info: boolean;
    keyboardControls: boolean;
    barColor: string;
    start: string;
    end: string;
}

export class BetterYouTube extends React.Component {
    static defaultProps = {
        width: 375,
        height: 196,
        youtubeID: "0eBqJkMabgI",
        autoplay: false,
        controls: "True",
        mute: false,
        info: false,
        keyboardControls: false,
        barColor: "red",
        start: "",
        end: "",
    };

    static propertyControls: PropertyControls<Props> = {
        youtubeID: { type: ControlType.String, title: "YouTube ID" },
        autoplay: { type: ControlType.Boolean, title: "Autoplay?" },
        controls: { type: ControlType.SegmentedEnum, title: "Controls?", options: ["True","False"] },
        mute: { type: ControlType.Boolean, title: "Mute?" },
        info: { type: ControlType.Boolean, title: "Info?" },
        keyboardControls: { type: ControlType.Boolean, title: "Keyboard?" },
        barColor: { type: ControlType.Enum, title: "Bar Color", options: ["red","white"] },
        start: { type: ControlType.String, title: "Start Time" },
        end: { type: ControlType.String, title: "End Time" },
    }

    render() {
        const width = Animatable.getNumber(this.props.width)
        const height = Animatable.getNumber(this.props.height)

        if (this.props.controls === "True") {
            var url = `https://www.youtube.com/embed/${this.props.youtubeID}?playlist=${this.props.youtubeID}&showinfo=${this.props.info}&controls=1&autoplay=${this.props.autoplay}&cc_load_policy=0&loop=${this.props.loop}&mute=${this.props.mute}&iv_load_policy=3&disablekb=${this.props.keyboardControls}&rel=0&color=${this.props.barColor}&start=${this.props.start}&end=${this.props.end}`;
        }
        else if (this.props.controls === "False") {
            var url = `https://www.youtube.com/embed/${this.props.youtubeID}?playlist=${this.props.youtubeID}&showinfo=${this.props.info}&controls=0&autoplay=${this.props.autoplay}&cc_load_policy=0&loop=${this.props.loop}&mute=${this.props.mute}&iv_load_policy=3&disablekb=${this.props.keyboardControls}&rel=0&color=${this.props.barColor}&start=${this.props.start}&end=${this.props.end}`;
        }

        return (
            <iframe width={width} height={height} frameBorder={0} src={url} />
        )
    }
}
