// @framer/toddjreynolds.elements-ui-kit/code/Acknowledgement.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType, Stack } from "framer"
import { Icon, iconNames } from "./Icon"
import { colors } from "./Colors"

// Learn more: https://framer.com/api

export function Acknowledgement(props) {
    const { title, text, background, color, onTap, ...rest } = props

    return (
        <Frame {...rest} background={null} onTap={onTap}>
            <Stack
                id="Container"
                gap={16}
                left={0}
                right={0}
                distribution="start"
                alignment="start"
                height="auto"
                width="auto"
                style={{
                    padding: "24px 0px 40px 0px",
                    borderRadius: "30px 30px 0 0",
                }}
                background={background}
            >
                <Stack
                    id="Title Group"
                    gap={8}
                    paddingLeft={24}
                    paddingRight={24}
                    distribution="start"
                    alignment="start"
                    height="auto"
                    width="100%"
                    background={null}
                >
                    <Frame
                        id="Title"
                        width="85%"
                        height="auto"
                        background={null}
                        opacity={0.88}
                        style={{
                            fontWeight: 600,
                            color: color,
                            fontSize: 17,
                            letterSpacing: -0.4,
                            lineHeight: 1.3,
                            textAlign: "left",
                        }}
                    >
                        <span>{title}</span>
                    </Frame>
                    <Frame
                        id="Text"
                        width="100%"
                        height="auto"
                        background={null}
                        opacity={0.7}
                        style={{
                            color: color,
                            fontSize: 16,
                            letterSpacing: -0.3,
                            lineHeight: 1.5,
                            textAlign: "left",
                        }}
                    >
                        <span>{text}</span>
                    </Frame>
                </Stack>
            </Stack>
            <Frame id="Icon" size={24} top={24} right={24} background="null">
                <Icon name="Close" tintColor={color} />
            </Frame>
            <Stack
                id="Home Indicator Bar"
                background={null}
                height={13}
                width="100%"
                bottom={0}
            >
                <Frame
                    background="#ffffff"
                    height={5}
                    width={134}
                    radius={2.5}
                />
            </Stack>
        </Frame>
    )
}

Acknowledgement.defaultProps = {
    height: "auto",
    width: 375,
    title: "Title",
    text:
        "This is where we heartily acknowledge the wisdom and grace of the user after they've successfully performed a task that we painstakingly oversimplified for them.",
    color: colors.black,
    background: colors.appTint,
}

// Learn more: https://framer.com/api/property-controls/
addPropertyControls(Acknowledgement, {
    title: {
        title: "Title",
        type: ControlType.String,
        defaultValue: "Witty Title Here",
    },
    text: {
        title: "Text",
        type: ControlType.String,
        displayTextArea: true,
        defaultValue:
            "This is where we heartily acknowledge the wisdom and grace of the user after they've successfully performed a task that we painstakingly oversimplified for them.",
    },
    background: {
        title: "Background",
        type: ControlType.Color,
        defaultValue: "#0099ff",
    },
    color: {
        title: "Content",
        type: ControlType.Color,
        defaultValue: colors.grayDark,
    },
    onTap: {
        type: ControlType.EventHandler,
    },
})
