// @framer/toddjreynolds.elements-ui-kit/code/AssetTile.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType, Stack } from "framer"
import { colors } from "./Colors"
import { Icon, iconNames } from "./Icon"
import { PrimaryButton } from "./PrimaryButton"

// Learn more: https://framer.com/api

export function AssetTile(props) {
    const { title, subtext, tint, name, onTap, ...rest } = props

    return (
        <Frame {...rest} background={null} onTap={onTap}>
            <Stack
                id="Container"
                gap={16}
                left={0}
                right={0}
                distribution="start"
                alignment="start"
                height="auto"
                width="auto"
                style={{
                    padding: "20px 0px 24px 0px",
                    borderRadius: 30,
                }}
                background={colors.tileBackgroundDark}
            >
                <Stack
                    id="Title Group"
                    gap={4}
                    paddingLeft={24}
                    paddingRight={24}
                    distribution="start"
                    alignment="start"
                    height="auto"
                    width="90%"
                    background={null}
                >
                    <Frame
                        id="Prompt Title"
                        width="auto"
                        height="auto"
                        background={null}
                        style={{
                            fontWeight: 700,
                            fontSize: 24,
                            color: colors.white,
                        }}
                    >
                        <span>{title}</span>
                    </Frame>
                    <Frame
                        id="Task Type"
                        width="auto"
                        height="auto"
                        background={null}
                        style={{
                            fontWeight: 400,
                            fontSize: 15,
                            color: colors.grayLight,
                        }}
                    >
                        <span>{props.showSubtext ? subtext : null}</span>
                    </Frame>
                </Stack>
                <Stack
                    id="Children"
                    width="100%"
                    height="auto"
                    background="null"
                >
                    {props.children}
                </Stack>
            </Stack>
            <Frame id="Icon" size={24} top={24} right={24} background="null">
                <Icon name={name} tintColor={tint} />
            </Frame>
        </Frame>
    )
}

AssetTile.defaultProps = {
    height: "auto",
    width: 343,
    title: "Title",
    showType: true,
    subtext: "Subtext",
    name: "Chevron Right",
    tint: colors.appTint,
}

// Learn more: https://framer.com/api/property-controls/
addPropertyControls(AssetTile, {
    title: {
        title: "Title",
        type: ControlType.String,
        defaultValue: "Title",
    },
    showSubtext: {
        title: "Subtext?",
        type: ControlType.Boolean,
        enabledTitle: "Yup",
        disabledTitle: "Nope",
        defaultValue: true,
    },
    subtext: {
        title: "Subtext",
        type: ControlType.String,
        defaultValue: "Subtext",
        displayTextArea: true,
        hidden(props) {
            return props.showType === false
        },
    },
    name: {
        title: "Icon",
        type: ControlType.Enum,
        options: iconNames.sort(),
        defaultValue: "Chevron Right",
    },
    tint: {
        title: "Tint",
        type: ControlType.Color,
        defaultValue: colors.appTint,
    },
    children: {
        title: "Content",
        type: ControlType.ComponentInstance,
    },
    onTap: {
        type: ControlType.EventHandler,
    },
})
