// @framer/toddjreynolds.elements-ui-kit/code/DestructiveConfirmation.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType, Stack } from "framer"
import { colors } from "./Colors"
import { Icon, iconNames } from "./Icon"
import { PrimaryButton } from "./PrimaryButton"

// Learn more: https://framer.com/api

export function DestructiveConfirmation(props) {
    const { title, tint, text, action, ...rest } = props

    return (
        <Stack
            {...rest}
            background={colors.grayDark}
            distribution="start"
            height="auto"
            gap={0}
            alignment="center"
            overflow="visible"
            style={{
                shadow: "0 2px 20px rgba(0,0,0,0.5)",
                boxShadow: "inset 0 1px 1px 0 rgba(255, 255, 255, 0.12)",
                borderRadius: "30px 30px 0 0",
            }}
        >
            <Frame width="100%" height={60} background="null">
                <Frame
                    id="Prompt Title"
                    width="auto"
                    height="auto"
                    top={20}
                    left={24}
                    right={56}
                    background={null}
                    style={{
                        fontWeight: 700,
                        fontSize: 24,
                        color: colors.white,
                    }}
                >
                    <span>{title}</span>
                </Frame>
                <Frame
                    id="Icon"
                    size={24}
                    top={24}
                    right={24}
                    background="null"
                >
                    <Icon name="Close" tintColor={tint} />
                </Frame>
            </Frame>
            <Frame
                id="Text"
                width="100%"
                height="auto"
                style={{
                    textAlign: "left",
                    color: colors.grayLight,
                    fontSize: 14,
                    lineHeight: 1.5,
                    paddingLeft: 24,
                    paddingRight: 24,
                }}
                background={null}
            >
                {text}
            </Frame>
            <Frame id="Spacer" height={32} background={null} />
            <Stack width="100%" height={56} paddingRight={32} paddingLeft={32}>
                <PrimaryButton buttonType="c" label={action} />
            </Stack>
            <Frame
                id="Bottom Spacing"
                height={66}
                width={375}
                bottom={0}
                left={0}
                right={0}
                background="null"
            >
                <Frame
                    id="Home Indicator"
                    background="#ffffff"
                    height={5}
                    width={134}
                    radius={2.5}
                    center="x"
                    bottom={8}
                />
            </Frame>
        </Stack>
    )
}

DestructiveConfirmation.defaultProps = {
    height: "auto",
    width: 375,
    title: "Are you sure?",
    tint: colors.appTint,
    text:
        "Aenean lacinia bibendum nulla sed consectetur. Maecenas faucibus mollis interdum. Etiam porta sem malesuada magna mollis euismod. Maecenas sed diam eget risus varius blandit sit amet non magna.",
    action: "Destroy Item",
}

// Learn more: https://framer.com/api/property-controls/
addPropertyControls(DestructiveConfirmation, {
    title: {
        title: "Title",
        type: ControlType.String,
        defaultValue: "Menu Title",
    },
    text: {
        title: "Text",
        type: ControlType.String,
        defaultValue:
            "Aenean lacinia bibendum nulla sed consectetur. Maecenas faucibus mollis interdum. Etiam porta sem malesuada magna mollis euismod. Maecenas sed diam eget risus varius blandit sit amet non magna. Sed posuere consectetur est at lobortis. Praesent commodo cursus magna, vel scelerisque nisl consectetur et. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.",
    },
    action: {
        title: "Action",
        type: ControlType.String,
        defaultValue: "Delete Me",
    },
    tint: {
        title: "Tint",
        type: ControlType.Color,
        defaultValue: colors.appTint,
    },
})
