// @framer/toddjreynolds.elements-ui-kit/code/ElementsOverviewList.tsx
import * as React from "react"
import {
    Frame,
    addPropertyControls,
    ControlType,
    Stack,
    Override,
} from "framer"
import { Icon, iconNames } from "./Icon"
import { colors } from "./Colors"
import { SectionHeader } from "./canvas"

// Learn more: https://framer.com/api

export function ElementsOverviewList(props) {
    const { showEstate, showIncome, showTaxes, ...rest } = props

    return (
        <Frame background={null} width="100%" height="auto">
            <Stack
                id="Main Stack"
                {...rest}
                background={colors.black}
                paddingTop={16}
                paddingBottom={66}
                gap={32}
                height="auto"
                style={{
                    color: "#fff",
                    fontSize: 18,
                    fontWeight: 400,
                }}
            >
                <Stack
                    id="Income Group"
                    width={375}
                    height="auto"
                    background={null}
                    gap={12}
                    radius={30}
                    overflow="hidden"
                >
                    <SectionHeader Label="INCOME" width={375} />
                    <Stack
                        id="Income Elements"
                        width={343}
                        height="auto"
                        background={colors.tileBackgroundDark}
                        gap={0}
                        radius={30}
                        overflow="hidden"
                    >
                        <Frame
                            id="Elements Item"
                            height={64}
                            width="100%"
                            background="null"
                        >
                            <Frame
                                background={null}
                                size={24}
                                top={20}
                                left={16}
                            >
                                <Icon
                                    name="Piggy Bank"
                                    tintColor={colors.savings}
                                />
                            </Frame>
                            <Frame
                                height={64}
                                width="auto"
                                left={56}
                                background="null"
                            >
                                Savings
                            </Frame>
                            <Frame
                                background="null"
                                size={24}
                                top={20}
                                right={12}
                            >
                                <Icon
                                    name="Chevron Right"
                                    tintColor={colors.grayMedium}
                                />
                            </Frame>
                            <Frame
                                id="Separator"
                                left={56}
                                right={0}
                                bottom={0}
                                height={1}
                                background={colors.white}
                                opacity={0.08}
                            />
                        </Frame>
                        <Frame
                            id="Elements Item"
                            height={64}
                            width="100%"
                            background="null"
                        >
                            <Frame
                                background={null}
                                size={24}
                                top={20}
                                left={16}
                            >
                                <Icon
                                    name="Credit Card"
                                    tintColor={colors.spending}
                                />
                            </Frame>
                            <Frame
                                height={64}
                                width="auto"
                                left={56}
                                background="null"
                            >
                                Spending
                            </Frame>
                            <Frame
                                background="null"
                                size={24}
                                top={20}
                                right={12}
                            >
                                <Icon
                                    name="Chevron Right"
                                    tintColor={colors.grayMedium}
                                />
                            </Frame>
                            <Frame
                                id="Separator"
                                left={56}
                                right={0}
                                bottom={0}
                                height={1}
                                background={colors.white}
                                opacity={0.08}
                            />
                        </Frame>
                        <Frame
                            id="Elements Item"
                            height={64}
                            width="100%"
                            background="null"
                        >
                            <Frame
                                background={null}
                                size={24}
                                top={20}
                                left={16}
                            >
                                <Icon name="Waves" tintColor={colors.debt} />
                            </Frame>
                            <Frame
                                height={64}
                                width="auto"
                                left={56}
                                background="null"
                            >
                                Debt
                            </Frame>
                            <Frame
                                background="null"
                                size={24}
                                top={20}
                                right={12}
                            >
                                <Icon
                                    name="Chevron Right"
                                    tintColor={colors.grayMedium}
                                />
                            </Frame>
                            <Frame
                                id="Separator"
                                left={56}
                                right={0}
                                bottom={0}
                                height={1}
                                background={colors.white}
                                opacity={0}
                            />
                        </Frame>
                    </Stack>
                </Stack>
                // -------------------------------------- MARK: ASSETS GROUP
                <Stack
                    id="Assets Group"
                    width={375}
                    height="auto"
                    background={null}
                    gap={12}
                    radius={30}
                    overflow="hidden"
                >
                    <SectionHeader Label="ASSETS" width={375} />
                    <Stack
                        id="Income Elements"
                        width={343}
                        height="auto"
                        background={colors.tileBackgroundDark}
                        gap={0}
                        radius={30}
                        overflow="hidden"
                    >
                        <Frame
                            id="Elements Item"
                            height={64}
                            width="100%"
                            background="null"
                        >
                            <Frame
                                background={null}
                                size={24}
                                top={20}
                                left={16}
                            >
                                <Icon
                                    name="Droplet"
                                    tintColor={colors.liquidity}
                                />
                            </Frame>
                            <Frame
                                height={64}
                                width="auto"
                                left={56}
                                background="null"
                            >
                                Liquidity
                            </Frame>
                            <Frame
                                background="null"
                                size={24}
                                top={20}
                                right={12}
                            >
                                <Icon
                                    name="Chevron Right"
                                    tintColor={colors.grayMedium}
                                />
                            </Frame>
                            <Frame
                                id="Separator"
                                left={56}
                                right={0}
                                bottom={0}
                                height={1}
                                background={colors.white}
                                opacity={0.08}
                            />
                        </Frame>
                        <Frame
                            id="Elements Item"
                            height={64}
                            width="100%"
                            background="null"
                        >
                            <Frame
                                background={null}
                                size={24}
                                top={20}
                                left={16}
                            >
                                <Icon
                                    name="Lock"
                                    tintColor={colors.retirement}
                                />
                            </Frame>
                            <Frame
                                height={64}
                                width="auto"
                                left={56}
                                background="null"
                            >
                                Retirement Accounts
                            </Frame>
                            <Frame
                                background="null"
                                size={24}
                                top={20}
                                right={12}
                            >
                                <Icon
                                    name="Chevron Right"
                                    tintColor={colors.grayMedium}
                                />
                            </Frame>
                            <Frame
                                id="Separator"
                                left={56}
                                right={0}
                                bottom={0}
                                height={1}
                                background={colors.white}
                                opacity={0.08}
                            />
                        </Frame>
                        <Frame
                            id="Elements Item"
                            height={64}
                            width="100%"
                            background="null"
                        >
                            <Frame
                                background={null}
                                size={24}
                                top={20}
                                left={16}
                            >
                                <Icon
                                    name="Home"
                                    tintColor={colors.realEstate}
                                />
                            </Frame>
                            <Frame
                                height={64}
                                width="auto"
                                left={56}
                                background="null"
                            >
                                Real Estate
                            </Frame>
                            <Frame
                                background="null"
                                size={24}
                                top={20}
                                right={12}
                            >
                                <Icon
                                    name="Chevron Right"
                                    tintColor={colors.grayMedium}
                                />
                            </Frame>
                            <Frame
                                id="Separator"
                                left={56}
                                right={12}
                                bottom={0}
                                height={1}
                                background={colors.white}
                                opacity={0.08}
                            />
                        </Frame>
                        <Frame
                            id="Elements Item"
                            height={64}
                            width="100%"
                            background="null"
                        >
                            <Frame
                                background={null}
                                size={24}
                                top={20}
                                left={16}
                            >
                                <Icon
                                    name="Storefront"
                                    tintColor={colors.business}
                                />
                            </Frame>
                            <Frame
                                height={64}
                                width="auto"
                                left={56}
                                background="null"
                            >
                                Practice
                            </Frame>
                            <Frame
                                background="null"
                                size={24}
                                top={20}
                                right={12}
                            >
                                <Icon
                                    name="Chevron Right"
                                    tintColor={colors.grayMedium}
                                />
                            </Frame>
                            <Frame
                                id="Separator"
                                left={56}
                                right={0}
                                bottom={0}
                                height={1}
                                background={colors.white}
                                opacity={0.08}
                            />
                        </Frame>
                        <Frame
                            id="Elements Item"
                            height={64}
                            width="100%"
                            background="null"
                        >
                            <Frame
                                background={null}
                                size={24}
                                top={20}
                                left={16}
                            >
                                <Icon
                                    name="Beach Umbrella"
                                    tintColor={colors.financialIndependence}
                                />
                            </Frame>
                            <Frame
                                height={64}
                                width="auto"
                                left={56}
                                background="null"
                            >
                                Financial Independence
                            </Frame>
                            <Frame
                                background="null"
                                size={24}
                                top={20}
                                right={12}
                            >
                                <Icon
                                    name="Chevron Right"
                                    tintColor={colors.grayMedium}
                                />
                            </Frame>
                            <Frame
                                id="Separator"
                                left={56}
                                right={0}
                                bottom={0}
                                height={1}
                                background={colors.white}
                                opacity={0}
                            />
                        </Frame>
                    </Stack>
                </Stack>
                // -------------------------------------- MARK: RISK GROUP
                <Stack
                    id="Risk Group"
                    width={375}
                    height="auto"
                    background={null}
                    gap={0}
                    radius={30}
                    overflow="hidden"
                >
                    <SectionHeader Label="RISK" width={375} />
                    <Stack
                        id="Income Elements"
                        width={343}
                        height="auto"
                        background={colors.tileBackgroundDark}
                        gap={12}
                        radius={30}
                        overflow="hidden"
                    >
                        <Frame
                            id="Elements Item"
                            height={64}
                            width="100%"
                            background="null"
                        >
                            <Frame
                                background={null}
                                size={24}
                                top={20}
                                left={16}
                            >
                                <Icon
                                    name="Shield (Verified)"
                                    tintColor={colors.insurance}
                                />
                            </Frame>
                            <Frame
                                height={64}
                                width="auto"
                                left={56}
                                background="null"
                            >
                                Insurance
                            </Frame>
                            <Frame
                                background="null"
                                size={24}
                                top={20}
                                right={12}
                            >
                                <Icon
                                    name="Chevron Right"
                                    tintColor={colors.grayMedium}
                                />
                            </Frame>
                            <Frame
                                id="Separator"
                                left={56}
                                right={12}
                                bottom={0}
                                height={1}
                                background={colors.white}
                                opacity={0.08}
                            />
                        </Frame>
                        <Frame
                            id="Elements Item"
                            height={64}
                            width="100%"
                            background="null"
                        >
                            <Frame
                                background={null}
                                size={24}
                                top={20}
                                left={16}
                            >
                                <Icon
                                    name="Chart Trending"
                                    tintColor={colors.investments}
                                />
                            </Frame>
                            <Frame
                                height={64}
                                width="auto"
                                left={56}
                                background="null"
                            >
                                Investments
                            </Frame>
                            <Frame
                                background="null"
                                size={24}
                                top={20}
                                right={12}
                            >
                                <Icon
                                    name="Chevron Right"
                                    tintColor={colors.grayMedium}
                                />
                            </Frame>
                            <Frame
                                id="Separator"
                                left={56}
                                right={12}
                                bottom={0}
                                height={1}
                                background={colors.white}
                                opacity={0}
                            />
                        </Frame>
                    </Stack>
                </Stack>
            </Stack>
        </Frame>
    )
}

ElementsOverviewList.defaultProps = {
    height: "auto",
    width: 375,
    showTaxes: false,
    showEstate: false,
    showIncome: false,
}

// Learn more: https://framer.com/api/property-controls/
addPropertyControls(ElementsOverviewList, {
    showIncome: {
        title: "Income",
        type: ControlType.Boolean,
        defaultValue: false,
    },
    showTaxes: {
        title: "Taxes",
        type: ControlType.Boolean,
        defaultValue: false,
    },
    showEstate: {
        title: "Estate",
        type: ControlType.Boolean,
        defaultValue: false,
    },
})
