// @framer/toddjreynolds.elements-ui-kit/code/FABButton.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType } from "framer"
import { Icon, iconNames } from "./Icon"

// Learn more: https://framer.com/api

export function FABButton(props) {
    const { text, tint, onTap, ...rest } = props

    return (
        <Frame
            {...rest}
            borderRadius={28}
            top={0}
            left={0}
            style={{
                WebkitBackdropFilter: "blur(6px)",
                backdropFilter: "blur(6px)",
                backgroundColor: "rgba(0,0,0,0)",
            }}
        >
            <Frame
                {...rest}
                background={tint}
                onTap={onTap}
                opacity={0.1}
                borderRadius={28}
                style={{}}
            />
            <Frame
                {...rest}
                background="null"
                borderWidth={1.5}
                borderColor={tint}
                style={{
                    borderRadius: 28,
                }}
            >
                <Icon name={props.name} tintColor={props.tint} />
            </Frame>
        </Frame>
    )
}

FABButton.defaultProps = {
    height: 56,
    width: 56,
    name: "Add",
    tint: "#4af196",
}

// Learn more: https://framer.com/api/property-controls/
addPropertyControls(FABButton, {
    name: {
        type: ControlType.Enum,
        title: "Icon",
        options: iconNames.sort(),
        defaultValue: "Add",
    },
    tint: {
        title: "Tint",
        type: ControlType.Color,
        defaultValue: "#4af196",
    },
    onTap: {
        type: ControlType.EventHandler,
    },
})
