// @framer/toddjreynolds.elements-ui-kit/code/LearnMoreButton.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType, Stack } from "framer"
import { Icon, iconNames } from "./Icon"

// Learn more: https://framer.com/api

export function LearnMoreButton(props) {
    const { text, tint, onTap, ...rest } = props

    return (
        <Frame
            {...rest}
            background="null"
            onTap={onTap}
            style={{
                color: props.tint,
                fontSize: 17,
                fontWeight: 500,
            }}
        >
            <Stack
                direction="horizontal"
                distribution="center"
                gap={2}
                width="100%"
                height="100%"
            >
                <span>{text}</span>
                <Frame {...rest} width={24} background="null">
                    <Icon name={props.name} tintColor={props.tint} />
                </Frame>
            </Stack>
        </Frame>
    )
}

LearnMoreButton.defaultProps = {
    height: 24,
    width: "100%",
    text: "Learn More",
    tint: "#4af196",
    name: "Chevron Right",
}

// Learn more: https://framer.com/api/property-controls/
addPropertyControls(LearnMoreButton, {
    name: {
        type: ControlType.Enum,
        title: "Icon",
        options: iconNames.sort(),
    },
    text: {
        title: "Text",
        type: ControlType.String,
        defaultValue: "Learn More",
    },
    tint: {
        title: "Tint",
        type: ControlType.Color,
        defaultValue: "#4af196",
    },
    onTap: {
        type: ControlType.EventHandler,
    },
})
