// @framer/toddjreynolds.elements-ui-kit/code/NavigationBarSmallTitle.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType, Stack } from "framer"
import { Icon } from "./Icon"
// import { colors } from "./Colors"
import { colors } from "./canvas"

// Learn more: https://framer.com/api

export function NavigationBarSmallTitle(props) {
    const { title, tint, onTap, ...rest } = props

    const titleType = {
        true: (
            <Stack
                    id="Title with Subtitle"
                    {...rest}
                    gap={-2}
                    height={38}
                    bottom={4}
                    style={{
                        fontWeight: 800,
                        fontSize: 18,
                        lineHeight: 24,
                        color: "white",
                    }}
                    background="null"
                >
                    <Frame id="Title" height={24} top={0} width="auto" background={null}>{props.title}</Frame>
                    <Frame id="Subtitle" height={16} bottom={0} width="auto" background={null} style={{
                        fontWeight: 500,
                        color: colors["Gray Light"],
                        fontSize: 14,
                    }}>{props.subtitle}</Frame>
                </Stack>
        ), 
        false: (
            <Frame
                    id="Small Title"
                    {...rest}
                    height={38}
                    top={0}
                    style={{
                        fontWeight: 800,
                        fontSize: 18,
                        color: colors.White,
                    }}
                    background="null"
                >
                    {props.title}
                </Frame>
        )
    }

    const leftAction = {
        a: (
            <Frame
                id="Cancel"
                height={44}
                width="auto"
                left={16}
                top={0}
                style={{
                    fontSize: 17,
                    color: props.tint,
                    paddingTop: 10,
                }}
                background="null"
            >
                <span>Cancel</span>
            </Frame>
        ),
        b: (
            <Frame
                id="Back Arrow"
                height={40}
                width={24}
                left={16}
                top={0}
                background="null"
            >
                <Icon name="Back Arrow" tintColor={props.tint} />
            </Frame>
        ),
        c: (
            <Frame id="Profile Badge" size={32} left={20} top={4} background={props.badgeColor} style={{
                borderRadius: 16, fontWeight: 800, fontSize: 14, color: colors["Gray Dark"],
            }} >
                {props.initials}
            </Frame>
        ),
        d: null,
    }

    const rightAction = {
        a: null,
        b: (
            <Frame
                id="Save Disabled"
                height={44}
                width="auto"
                right={16}
                top={0}
                style={{
                    fontSize: 17,
                    fontWeight: 600,
                    color: "#585f66",
                    paddingTop: 10,
                }}
                background="null"
            >
                <span>Save</span>
            </Frame>
        ),
        c: (
            <Frame
                id="Save Enabled"
                height={44}
                width="auto"
                right={16}
                top={0}
                style={{
                    fontSize: 17,
                    fontWeight: 600,
                    color: props.tint,
                    paddingTop: 10,
                }}
                background="null"
            >
                <span>Save</span>
            </Frame>
        ),
        d: (
            <Frame
                id="Edit"
                height={44}
                width="auto"
                right={16}
                top={0}
                style={{
                    fontSize: 17,
                    color: tint,
                    paddingTop: 10,
                }}
                background="null"
            >
                <span>Edit</span>
            </Frame>
        ),
        e: (
            <Frame
                id="Close Button"
                height={40}
                width={24}
                right={16}
                top={0}
                background="null"
            >
                <Icon name="Close" tintColor={tint} />
            </Frame>
        ),
    }

    return (
        <Frame id="Navbar" {...rest} background={null}>
            <Frame
                id="Actions"
                height={44}
                width="100%"
                bottom={0}
                background={null}
            >
                {leftAction[props.leftAction]}
                {rightAction[props.rightAction]}
            </Frame>
            {titleType[props.showSubtitle]}
        </Frame>
    )
}

NavigationBarSmallTitle.defaultProps = {
    height: 88,
    width: 375,
    title: "Perry O'Donnell",
    tint: "#4af196",
    leftAction: "a",
    rightAction: "a",
    showSubtitle: "true",
    initials: "PO",
    badgeColor: colors.Income,
}

// Learn more: https://framer.com/api/property-controls/
addPropertyControls(NavigationBarSmallTitle, {
    leftAction: {
        title: "Left Action",
        type: ControlType.Enum,
        options: ["a", "b", "c", "d"],
        optionTitles: ["Cancel", "Back", "Profile Badge", "None"],
        defaultValue: "a",
    },
    rightAction: {
        title: "Right Action",
        type: ControlType.Enum,
        options: ["a", "b", "c", "d", "e"],
        optionTitles: [
            "None",
            "Save (Disabled)",
            "Save (Enabled)",
            "Edit",
            "Close",
        ],
        defaultValue: "a",
    },
    title: {
        title: "Title",
        type: ControlType.String,
        defaultValue: "View Title",
    },
    showSubtitle: {
        title: "Subtitle?", 
        type: ControlType.Boolean, 
        defaultValue: true,
        enabledTitle: "Yup",
        disabledTitle: "Nope"
    },
    subtitle: {
        title: "Subtitle",
        type: ControlType.String,
        defaultValue: "Subtitle Goes Here",
    },
    showInitials: {
        title: "Profile?",
        type: ControlType.Boolean,
        enabledTitle: "Yes",
        disabledTitle: "No"
    },
    initials: {
        type: ControlType.String, 
        title: "User Initials",
        defaultValue: "PO",
        hidden(props) {
            return props.showInitials === false
        }
    },
    badgeColor: {
        title: "Color",
        type: ControlType.Color, 
        defaultValue: colors.Income,
        hidden(props) {
            return props.showInitials === false
        }
    },
    tint: {
        title: "Tint",
        type: ControlType.Color,
        defaultValue: colors["Default Tint"],
    },
    onTap: {
        type: ControlType.EventHandler,
    },
})
