// @framer/toddjreynolds.elements-ui-kit/code/OptionsBottomSheet.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType, Stack } from "framer"
import { colors } from "./Colors"
import { Icon, iconNames } from "./Icon"
import { PrimaryButton } from "./PrimaryButton"
import { SelectionItem } from "./SelectionItem"

// Learn more: https://framer.com/api

export function OptionsBottomSheet(props) {
    const { title, tint, last, ...rest } = props

    const options = props.rows.map((label) => (
        <SelectionItem label={label} tint={tint} lastItem={false} />
    ))

    return (
        <Stack
            {...rest}
            background={colors.grayDark}
            distribution="start"
            height="auto"
            gap={0}
            alignment="center"
            overflow="visible"
            style={{
                shadow: "0 2px 20px rgba(0,0,0,0.5)",
                boxShadow: "inset 0 1px 1px 0 rgba(255, 255, 255, 0.12)",
                borderRadius: "30px 30px 0 0",
            }}
        >
            <Frame width="100%" height={80} background="null">
                <Frame
                    id="Prompt Title"
                    width="auto"
                    height="auto"
                    top={20}
                    left={24}
                    right={56}
                    background={null}
                    style={{
                        fontWeight: 700,
                        fontSize: 24,
                        color: colors.white,
                    }}
                >
                    <span>{title}</span>
                </Frame>
                <Frame
                    id="Icon"
                    size={24}
                    top={24}
                    right={24}
                    background="null"
                >
                    <Icon name="Close" tintColor={tint} />
                </Frame>
            </Frame>
            {options}
            <SelectionItem
                id="Last Option"
                label={last}
                tint={tint}
                lastItem={true}
            />
            <Frame
                id="Bottom Spacing"
                height={66}
                width={375}
                bottom={0}
                left={0}
                right={0}
                background="null"
            >
                <Frame
                    background="#ffffff"
                    height={5}
                    width={134}
                    radius={2.5}
                    center="x"
                    bottom={8}
                />
            </Frame>
        </Stack>
    )
}

OptionsBottomSheet.defaultProps = {
    height: "auto",
    width: 375,
    title: "Menu Title",
    tint: colors.business,
    rows: ["Option 1", "Option 2", "Option 3"],
    last: "Last Option",
}

// Learn more: https://framer.com/api/property-controls/
addPropertyControls(OptionsBottomSheet, {
    title: {
        title: "Title",
        type: ControlType.String,
        defaultValue: "Menu Title",
    },
    tint: {
        title: "Tint",
        type: ControlType.Color,
        defaultValue: colors.appTint,
    },
    rows: {
        title: "Options",
        type: ControlType.Array,
        propertyControl: {
            type: ControlType.String,
        },
        defaultValue: ["Item 1", "Item 2"],
        maxCount: 16,
    },
    last: {
        title: "Last Item",
        type: ControlType.String,
        defaultValue: "Last Item",
    },
    // child: {
    //     title: "Content",
    //     type: ControlType.ComponentInstance,
    // },
})
