// @framer/toddjreynolds.elements-ui-kit/code/PrimaryButton.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType } from "framer"
import { colors } from "./Colors"

// Learn more: https://framer.com/api

export function PrimaryButton(props) {
    const { label, tint, onTap, ...rest } = props
    const disabledColor = "#575757"
    const showButtonStyle = {
        // Default Button
        a: (
            <Frame
                id="Blur"
                {...rest}
                borderRadius={28}
                style={{
                    WebkitBackdropFilter: "blur(10px)",
                    backdropFilter: "blur(10px)",
                    backgroundColor: "rgba(0,0,0,0.32)",
                }}
            >
                <Frame
                    id="Tinted Background"
                    {...rest}
                    background={tint}
                    opacity={0.1}
                    borderRadius={28}
                />
                <Frame
                    id="Border and Label"
                    {...rest}
                    borderWidth={1.5}
                    borderColor={tint}
                    background={null}
                    style={{
                        color: tint,
                        fontSize: 16,
                        fontWeight: 600,
                        textTransform: "uppercase",
                        letterSpacing: 0.2,
                        borderRadius: 28,
                    }}
                >
                    {label}
                </Frame>
            </Frame>
        ),
        // Disabled Button
        b: (
            <Frame
                id="Blur"
                {...rest}
                borderRadius={28}
                style={{
                    WebkitBackdropFilter: "blur(10px)",
                    backdropFilter: "blur(10px)",
                    backgroundColor: "rgba(0,0,0,0.1)",
                }}
            >
                <Frame
                    id="Tinted Background"
                    {...rest}
                    background={disabledColor}
                    opacity={0.2}
                    borderRadius={28}
                />
                <Frame
                    id="Border and Label"
                    {...rest}
                    // borderRadius={28}
                    borderWidth={1.5}
                    borderColor={disabledColor}
                    style={{
                        color: disabledColor,
                        fontSize: 16,
                        fontWeight: 600,
                        textTransform: "uppercase",
                        letterSpacing: 0.2,
                        background: "transparent",
                        borderRadius: 28,
                    }}
                >
                    {label}
                </Frame>
            </Frame>
        ),
        // Destructive Button
        c: (
            <Frame
                id="Blur"
                {...rest}
                borderRadius={28}
                style={{
                    WebkitBackdropFilter: "blur(10px)",
                    backdropFilter: "blur(10px)",
                    backgroundColor: "rgba(0,0,0,0.1)",
                }}
            >
                <Frame
                    id="Tinted Background"
                    {...rest}
                    background={colors.business}
                    opacity={0.8}
                    borderRadius={28}
                />
                <Frame
                    id="Border and Label"
                    {...rest}
                    // borderRadius={28}
                    borderWidth={1.5}
                    borderColor={colors.business}
                    style={{
                        color: colors.grayDark,
                        fontSize: 16,
                        fontWeight: 600,
                        textTransform: "uppercase",
                        letterSpacing: 0.2,
                        background: "transparent",
                        borderRadius: 28,
                    }}
                >
                    {label}
                </Frame>
            </Frame>
        ),
    }

    return (
        // Background
        <Frame id="Container" width="100%" height={56} background={null}>
            {showButtonStyle[props.buttonType]}
        </Frame>
    )
}

PrimaryButton.defaultProps = {
    height: 56,
    width: 311,
    label: "Button",
    tint: "#4af196",
    buttonType: "a",
}

// Learn more: https://framer.com/api/property-controls/
addPropertyControls(PrimaryButton, {
    buttonType: {
        title: "Type",
        type: ControlType.Enum,
        options: ["a", "b", "c"],
        optionTitles: ["Default", "Disabled", "Destructive"],
        defaultValue: "a",
    },

    label: {
        title: "Label",
        type: ControlType.String,
        defaultValue: "Hello Framer!",
    },
    tint: {
        title: "Tint",
        type: ControlType.Color,
        defaultValue: "#0099ff",
        hidden(props) {
            return props.buttonType !== "a"
        },
    },
    onTap: {
        type: ControlType.EventHandler,
    },
})
