// @framer/toddjreynolds.elements-ui-kit/code/SelectionItem.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType, Stack } from "framer"
import { colors } from "./Colors"
import { Icon, iconNames } from "./Icon"
// Learn more: https://framer.com/api

export function SelectionItem(props) {
    const { label, tint, lastItem } = props

    const [selection, setSeletion] = React.useState(false)

    function toggleSelection() {
        if (selection === false) {
            setSeletion(true)
        } else setSeletion(false)
    }

    return (
        <Frame
            id="Container"
            height={56}
            width="100%"
            onTap={toggleSelection}
            background={null}
        >
            <Frame id="icon" size={24} center="y" left={16} background={null}>
                <Icon
                    name={
                        selection
                            ? "Checkmark (Filled)"
                            : "Radio Button (Empty)"
                    }
                    tintColor={selection ? tint : colors.grayMedium}
                />
            </Frame>
            <Frame
                id="label"
                height="auto"
                width="auto"
                center="y"
                left={64}
                right={16}
                background={null}
                style={{
                    fontSize: 18,
                    color: colors.white,
                }}
            >
                <span>{label}</span>
            </Frame>
            <Frame
                id="separator"
                height={1}
                left={64}
                right={16}
                bottom={0}
                background={colors.white}
                opacity={lastItem ? 0 : 0.08}
            />
        </Frame>
    )
}

SelectionItem.defaultProps = {
    height: 56,
    width: 375,
    label: "Option Label",
    tint: colors.appTint,
    lastItem: false,
}

// Learn more: https://framer.com/api/property-controls/
addPropertyControls(SelectionItem, {
    label: {
        title: "Label",
        type: ControlType.String,
        defaultValue: "Option Label",
    },
    tint: {
        title: "Tint",
        type: ControlType.Color,
        defaultValue: colors.appTint,
    },
    lastItem: {
        title: "Last?",
        type: ControlType.Boolean,
        defaultValue: false,
    },
})
